starter-code/routes/siteController.js

```diff
@@ -1,8 +1,118 @@
 const express = require("express");
-const siteController = express.Router();
+const router = express.Router();
+const User = require("../models/user");
+const mongoose = require("mongoose");
+mongoose.connect("mongodb://localhost/ibi-ironhack");
+const checkRoles = require("../middlewares/check-role");
+// Bcrypt to encrypt passwords
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
+const ensureLogin = require("connect-ensure-login");
 
-siteController.get("/", (req, res, next) => {
+const passport = require("passport");
+//define roles
+const checkBoss = checkRoles("Boss");
+const checkDeveloper = checkRoles("Developer");
+const checkTA = checkRoles("TA");
+
+//First iteration
+
+router.get("/", (req, res, next) => {
   res.render("index");
 });
 
-module.exports = siteController;
+router.get("/login", (req, res) => {
+  res.render("login", { message: req.flash("error") });
+});
+
+router.post(
+  "/login",
+  passport.authenticate("local", {
+    successRedirect: "/platform",
+    failureRedirect: "/login",
+    failureFlash: true,
+    passReqToCallback: true
+  })
+);
+
+router.get(
+  "/platform",
+  checkBoss,
+  ensureLogin.ensureLoggedIn(),
```

Comment:
Maybe it's better to check the user is logged in, before checking the role. It's more efficient, as you are stopping the pipeline if he's not logged in, without a need to check the role for that case